### DISCUSSION TITLE
Github Copilot logged into wrong account( VS2022 not VSC), can I change account?

### DISCUSSION THREAD

**Discussion by Nuyek** [community/community#13835] [Copilot Conversations] [answered]
Hello, I logged into the wrong account for copilot in VS 2022. I tried re-installing and clearing cache folders but still nothing. Is there a location to reset the details for the extension? Signing in keeps trying to go to the old account. Thank you.

I also tried revoking access inside the old github.

**Comment by dev-eth** [comment]
Change your primary email address in the GitHub settings.

**Reply by zombiezach8** [reply]
Tried that and it's still complaining that the account I signed in does not have copilot access. Trying to get it to use my other account but cannot find out where to reset the extension

**Reply by dev-eth** [reply]
Remove copilot from the VSCode extension list first. And after reinstalling, authorize again. (Make sure you cleared the browser cache history)
1 question - Did you already receive the invitation to the Github copilot Technical Preview?

**Reply by Nuyek** [reply]
Yes I have the invitation in the other account. Tried clearing cache after uninstalling the extension but I never get the option to authorize again, it's still trying to sign in with the old account.

**Reply by dev-eth** [reply]
Then you should contact the Github team.
As you saw when you signup the Copilot waitlist, they said you need to make sure your primary email is the correct one that you are using.
If you already got the invitation to the other account, you must use it.
Or contact the GitHub team, let's see how they reply.
Cheers

**Comment by kzu** [comment]
In Visual Studio, you can click Sign Out after you signed in (perhaps with the wrong account). At this point, when VS navigates to the device authorization page, it's the browser's responsibility to determine which account it will use. You can navigate yourself on your default browser to github.com and explicitly sign out of the website. This way, when the extension navigates to the device authorization page, you will get a chance to sign-in again to GitHub with the right account.

Please let me know how it goes!

**Reply by Nuyek** [reply]
The option to sign out does not exist, only sign in. When I try hitting sign in it still is trying to use the old account, I have tried clearing the cookies for github (and reinstalling VS/deleting cache). I never get the option to try using a different account or browser login.

**Reply by kzu** [reply]
What do you mean it's "trying to use the old account" if you never get the browser navigated to the device auth code? The device auth page is https://github.com/login/device and when you navigate to it in the default browser, if you're already authenticated, it will use whichever account you authenticated with. If nothing is navigating, I suggest you launch VS with `devenv /log` and see if there's additional troubleshooting info in the extra logging you'll get. 

VPN usage is a known issue at this time.

**Comment by TobiasLjunggren** [comment]
Same issue for me. After installing, clearing web browser, it still says I don't have access. No alternative to log out or change account

**Reply by dev-eth** [reply]
Did you check the VS Code version? Make sure you are using the latest one.

**Reply by dev-eth** [reply]
Copilot does not support the old veresion.

**Reply by TobiasLjunggren** [reply]
I solved it by using the Copilot Nightly version instead, but I am using the latest version of VSCode

**Reply by dev-eth** [reply]
I see.

**Reply by kzu** [reply]
There is no such thing as a Copilot Nightly in the VS2022 (which is what this discussion is about)...

**Comment by thedatarat** [comment]
Yeah I was only able to resolve this by using Nightly. Was able to download the regular extension on a different laptop tho.

**Comment by rsydstudent** [comment]
I try with this solution and work [https://github.com/community/community/discussions/7231#discussioncomment-2085202](url)

**Reply by david-tucu-gh1** [reply]
The url returns 404 (Page Not Found)

**Comment by sunnybak** [comment]
This works for me: https://github.com/community/community/discussions/6823

**Reply by JustBnLazy** [reply]
I was having the same issue. I went into the incorrect account, on GitHub, and revoked access. Then restarted VS 2022 and it prompted me again and then I made sure to log into the correct account. All is right with the world again!

**Reply by garrm** [reply]
FYI: @shiksb solution is for VSCode not VS2022. Why this is so hard is beyond me.

**Comment by Francois-BONNIN** [comment]
https://github.com/orgs/community/discussions/6823#discussioncomment-1577174 
It's work !👌

**Reply by morris-bah** [reply]
Again, this "solution" is for Visual Studio **Code**, not Visual Studio 2022!
Visual Studio 2022 doesn't have the icon referenced in the VS Code solution(s)
We need a solution that works for Visual Studio 2022!

**Reply by JustBnLazy** [reply]
Hi @morris-bah , did you see my post above where I was able to "reset" the account once I revoked access, for incorrect account, on GitHub?

**Comment by morris-bah** [comment]
I saw that, but I don't see any way to revoke access when logged into the incorrect account.
Under Settings, Copilot only offers me the option to sign up for a free trial

**Comment by morris-bah** [comment]
Oh, I may have just found it under Applications in Settings

**Comment by morris-bah** [comment]
So that worked. The Revoke item is in Settings under Applications.
![image](https://user-images.githubusercontent.com/106198926/227031938-2e7ad3b8-6928-4207-a6ec-849b3cfc4d4e.png)

**Comment by ahsinanwar** [comment]
You need to make sure, your loggedin account in visual studio and copilot chat preview account should be same.

**Comment by LTCat2** [comment]
Delete the two files (hosts.json and terms.json) in C:\Users\%USERNAME%\AppData\Local\github-copilot

**Reply by DanielLaubeABB** [reply]
on my side it was just hosts.json, but works perfectly for VS2022!

**Comment by robindegen** [comment]
Revoking copilot on the other account worked for me. This is terribly confusing. I should just have an option to log out from visual studio.

**Comment by ttbmngt** [comment]
There is an option to log out from VSCode. Click on the account icon bottom left to log out. You'll have to sign out of the github account you're currently logged into to switch accounts. 

<img width="506" alt="Screenshot 2023-07-02 at 2 53 12 PM" src="https://github.com/community/community/assets/135276812/3f8737d8-5ebe-412e-8ef9-d4705626871c">

**Reply by robindegen** [reply]
Yep but there is no such thing in Visual Studio 2022. VSCode is completely different, and the fact that microsoft named it something with "Visual Studio" has made googling for problems really difficult and annoying.

**Comment by Joel-play-lu** [comment]
I had the same issue with **VS2022** because my github credential was saved in sourceTree. 
Once I removed the credential from sourceTree I could click on the copilot icon in **VS2022** and choose Add Account.

Hope it helps

**Comment by yeucodonvn** [comment, marked as answer]
VS 2022
File -> account setting -> remove old account and link to new account

**Reply by robindegen** [reply]
That was changed more recently. A year ago it didn't show up there. So this is an improvement; albeit a bit convoluted
